src/go/tests/relay/in_process_relay_test.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/glog"
+
+	"github.com/googlecloudrobotics/core/src/go/cmd/http-relay-client/client"
+	"github.com/googlecloudrobotics/core/src/go/cmd/http-relay-server/server"
+)
+
+var (
+	port      = 8081
+	blockSize = 10 * 1024
+	once      sync.Once
+)
+
+func initRelay() {
+	once.Do(func() {
+		glog.Info("Running init.")
+
+		go func() {
+			relayServer := server.NewServer()
+			relayServer.Start(port, blockSize)
```

Comment:
`nok8s_relay_test.go` let's the OS auto assign a free port, it then scans stdout to learn which port is used. I think we can improve on it here by using port `0` and getting the actual port used after the server is started.

The catch is that `relayServer.port` will not contain the actual port used in this case, we need to update it after we start listening. https://github.com/googlecloudrobotics/core/blob/41c3e73120d1a431e699c798f36eece275e54952/src/go/cmd/http-relay-server/server/server.go#L534